migration/src/m0000530_base_purl_index.rs

```diff
@@ -0,0 +1,72 @@
+use sea_orm_migration::prelude::*;
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[async_trait::async_trait]
+#[allow(deprecated)]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .get_connection()
+            .execute_unprepared(
+                r#"
+                CREATE EXTENSION IF NOT EXISTS pg_trgm;
+                CREATE INDEX BasePurlTypeGinIdx ON base_purl USING GIN ((type) gin_trgm_ops);
```

Comment:
I guess SeaORM doesn't provide enough API to do this type-safely instead of stringy?

Worth a PR upstream (separately; not blockingly)?

Reply:
yes I found nothing stable for create extensions or referring to **gin_trgm_ops** ... will attempt PR upstream

Reply:
I think there some options, not sure they lead to the required result:

* https://docs.rs/sea-query/latest/sea_query/index/struct.IndexCreateStatement.html#method.full_text
* https://docs.rs/sea-query/latest/sea_query/index/struct.IndexCreateStatement.html#method.index_type

Reply:
Yes, full_text() just creates a 'plain ole' GIN index ... I was not able to use or create a GIN index with **gin_trgm_ops** using anything sea_orm provides ... we also probably need to research if we need tsquery/full text search generically across trustify at some point.